tests/unit/test_lr_schedulers.py

```diff
@@ -205,3 +224,345 @@ def _test_lr_warmup_decay_schedule(args,
+                              10),
+                             (1e-3,
+                              1e-1,
+                              0,
+                              21),
+                         ])
+def test_onecycle_lr(tmpdir, min_lr, max_lr, decay_rate, step_size):
+    config_dict = {
+        "train_batch_size": 2,
+        "steps_per_print": 1,
+        "optimizer": {
+            "type": "Adam",
+            "params": {
+                "lr": 0.00015
+            },
+        },
+        "scheduler": {
+            "type": ONE_CYCLE,
+            "params": {
+                CYCLE_MIN_LR: min_lr,
+                CYCLE_MAX_LR: max_lr,
+                DECAY_LR_RATE: decay_rate,
+                CYCLE_FIRST_STEP_SIZE: step_size,
+                DECAY_STEP_SIZE: step_size
+            }
+        },
+        "gradient_clipping": 1.0
+    }
+    args = args_from_dict(tmpdir, config_dict)
+    hidden_dim = 10
+
+    model = SimpleModel(hidden_dim, empty_grad=False)
+
+    @distributed_test(world_size=[1])
+    def _test_onecycle_lr(args,
+                          model,
+                          hidden_dim,
+                          min_lr,
+                          max_lr,
+                          step_size,
+                          decay_rate):
+        model, _, _, lr_scheduler = deepspeed.initialize(args=args,
+                                                         model=model,
+                                                         model_parameters=model.parameters())
+        data_loader = random_dataloader(model=model,
+                                        total_samples=max(50,
+                                                          step_size * 3),
+                                        hidden_dim=hidden_dim,
+                                        device=model.device,
+                                        dtype=torch.float)
+
+        step_lrs = []
+        for _, batch in enumerate(data_loader):
+            step_lrs.append(lr_scheduler.get_lr())
+            loss = model(batch[0], batch[1])
+            model.backward(loss)
+            model.step()
+
+        # Verify starting lr
+        assert step_lrs[0] == min_lr
+
+        # Verify peak lr
+        assert step_lrs[step_size] == max_lr
+
+        # Verify increasing phase
+        _verify_continuous_increase(step_lrs[:step_size])
+
+        # Verify decreasing phase
+        _verify_continuous_decrease(step_lrs[step_size:(step_size * 2)])
+
+        # Verify decay phase
+        if decay_rate > 0:
+            _verify_continuous_decrease(step_lrs[(step_size * 2):])
+
+    _test_onecycle_lr(args=args,
+                      model=model,
+                      hidden_dim=hidden_dim,
+                      min_lr=[min_lr],
+                      max_lr=[max_lr],
+                      step_size=step_size,
+                      decay_rate=decay_rate)
+
+
+@pytest.mark.parametrize("min_mom, max_mom, decay_rate, step_size",
+                         [
```

Comment:
The code formatter is really wrecking your parametrization, sorry :-(. We should probably revisit its config as its line breaks are often pretty bad.

You may want to append ` # yapf: disable` to the expression to skip formatting. You can see the same thing in `test_config.py` and other places.

Reply:
Done. Thanks!